### ISSUE TITLE
Output of evaluate and fit are not consistent

### ISSUE THREAD

**Issue by benjaminklein** [fchollet/keras#503]
I'm running the cifar_cnn example. I'm not using data_augmentation (changed the flag to False) and also I'm only training for 10 epochs.

After 10 epochs, I'm computing the loss on the **train** using the evaluate function of the method. The loss printed by the fit function at the end of the 10th epoch is much higher than the error on train when using evaluate.

Can someone please help me to understand what's wrong?

Thank you!

**Comment by fchollet** [comment]
`fit` and `evaluate` are measuring completely different quantities (`fit` monitors the average of the loss during training while `evaluate` looks at the loss of a static network, after training).

**Comment by benjaminklein** [comment]
Can you explain more on the measuring of fit? the print on each epoch is
the mean loss from the beginning of the training or from the beginning of
that epoch?

**Comment by fchollet** [comment]
From the beginning of the epoch. It's the sigma of the loss of each batch times the number of samples in the batch divided by the total number of samples, for the batches in the epoch considered.

**Comment by benjaminklein** [comment]
Please see the output below. At the end of 10th epoch I'm getting a train
error of 0.5850
Then using evaluate I'm checking the error on the train and I'm getting
0.371. While I understand that there should be a difference between the
loss at the end of 10th epoch (which is 0.5850) it's still very far from
the loss when calling evaluate (which is 0.371).


Epoch 0
50000/50000 [==============================] - 23s - loss: 1.6584
Epoch 1
50000/50000 [==============================] - 23s - loss: 1.2406
Epoch 2
50000/50000 [==============================] - 23s - loss: 1.0426
Epoch 3
50000/50000 [==============================] - 23s - loss: 0.9223
Epoch 4
50000/50000 [==============================] - 23s - loss: 0.8315
Epoch 5
50000/50000 [==============================] - 23s - loss: 0.7670
Epoch 6
50000/50000 [==============================] - 24s - loss: 0.7101
Epoch 7
50000/50000 [==============================] - 24s - loss: 0.6657
Epoch 8
50000/50000 [==============================] - 24s - loss: 0.6241
Epoch 9
50000/50000 [==============================] - 24s - loss: 0.5850
50000/50000 [==============================] - 6s
Train Score 0.371491642175

**Comment by fchollet** [comment]
One more thing: with `train` the network is in train mode (Dropout ON), with `evaluate` it is in test mode (Dropout OFF). Hence big differences.

**Comment by benjaminklein** [comment]
Makes sense. Thank you.
